src/docker-osx-dev

```diff
@@ -371,14 +371,13 @@ function init_boot2docker {
     boot2docker init
   fi
 
-  check_for_shared_folders
-
   if ! is_boot2docker_running; then
     log_info "Starting Boot2Docker VM"
     boot2docker start --vbox-share=disable
   fi
 
   configure_boot2docker
+  check_for_shared_folders
```

Comment:
I may be mis-remembering, but I believe you need to remove shared folders *before* running `boot2docker start`. Therefore, instead of moving the `_check_for_shared_folders` call to the end, I think it might be better to move the `configure_boot2docker` call towards the beginning.